src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/Router.js

```diff
@@ -39,46 +42,65 @@ export default class Router {
                 parameters[names[i - 1].name] = match[i];
             }
 
-            this.navigate(name, parameters);
+            const URLSearchParameters = new URLSearchParams(queryString);
+            const searchParameters = {};
+            for (const [key, value] of URLSearchParameters) {
+                searchParameters[key] = value;
+            }
+
+            this.navigate(name, parameters, searchParameters);
 
             break;
         }
     }
 
-    @action navigate(name: string, parameters: Object = {}) {
+    @action navigate(name: string, parameters: Object = {}, searchParameters: Object = {}) {
         const currentRoute = routeStore.get(name);
         const currentParameters = {...currentRoute.parameters, ...parameters};
+        const currentSearchParameters = searchParameters;
 
         if (this.currentRoute
             && currentRoute
             && this.currentRoute.name === currentRoute.name
-            && this.currentParameters
-            && currentParameters
-            && this.currentParameters.length === currentParameters.length
+            && this.compareParameters(this.currentParameters, currentParameters)
+            && this.compareParameters(this.currentSearchParameters, currentSearchParameters)
         ) {
-            let match = true;
+            return;
+        }
+
+        this.currentRoute = currentRoute;
+        this.currentParameters = currentParameters;
+        this.currentSearchParameters = currentSearchParameters;
+    }
 
-            for (let key in currentParameters) {
-                if (this.currentParameters[key] !== currentParameters[key]) {
-                    match = false;
-                    break;
-                }
+    compareParameters(currentParameters: Object, parameters: Object) {
```

Comment:
Instead of this function we should probably use a library which compares different objects by equality. We could e.g. use [_.isEquals](https://lodash.com/docs/4.17.4#isEqual). There is even a [package](https://www.npmjs.com/package/lodash.isequal) only containing this method.

Another solution would be to implement that on our own (could be basically this function). Would have the advantage that we are less dependent. But I think using the `lodash.isEqual` package would be fine, since it is not only a left-pad implementation 😉 

/cc @chirimoya @3rfan

Reply:
add dependency